src/Content/Conversation/Repository/Channel.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Content\Conversation\Repository;
+
+use Friendica\BaseCollection;
+use Friendica\Content\Conversation\Entity\Timeline as TimelineEntity;
+use Friendica\Content\Conversation\Factory\Timeline;
+use Friendica\Database\Database;
+use Psr\Log\LoggerInterface;
+
+class Channel extends \Friendica\BaseRepository
+{
+	protected static $table_name = 'channel';
+
+	public function __construct(Database $database, LoggerInterface $logger, Timeline $factory)
```

Comment:
This class should instantiate `UserCreatedChannelTimeline` with a dedicated `UserCreatedChannelTimelineFactory`.

Reply:
So you mean that I should rename this class to `src/Content/Conversation/Repository/UserCreatedChannelTimeline.php`? And which functions of this class here should be moved to the factory? I had got the impression that all database functions belong to this repository thingy and the more abstract ones belong to the factory?

Reply:
I don't know, what I have to code when you say "the Repository through the Factory should start instantiating UserCreatedChannelTimeline entities, which warrants the renaming.".

Reply:
It should look something like this:
```php
namespace Friendica\Content\Conversation\Repository;
namespace Friendica\Content\Conversation\Factory;

class UserCreatedChannelTimeline extends BaseRepository
{
    public function __construct(Database $database, LoggerInterface $logger, Factory\UserCreatedChannelTimeline $factory) {
```
The new `UserCreatedChannelTimeline` factory will have methods to instantiate the new `UserCreatedChannelTimeline` entity, including the `createFromTableRow` method declared in the interface `ICanCreateFromTableRow` that it should implement.

Reply:
I now did something. I'm unsure, if this meets your expectancy.

Reply:
So this works because the `createFromTableRow` method was defined in `Factory\Timeline`, but that's not where it should be. The only factory who should be able to create entities from table rows is `Factory\UserDefinedChannel`.

Summary of the changes:
- Move `Factory\Timeline->createFromTableRow` to `Factory\UserDefinedChannel`
- Change dependency in this constructor to `Factory\UserDefinedChannel`
- Rename to `Repository\UserDefinedChannel` for consistency.